frontend/hooks/useService.ts

```diff
@@ -1,53 +1,82 @@
 import { useMemo } from 'react';
 
+import { MiddlewareDeploymentStatus } from '@/client';
+import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
 import { Address } from '@/types/Address';
-import { Service } from '@/types/Service';
+
+import { useServices } from './useServices';
+
+type ServiceChainIdAddressRecord = {
+  [chainId: number]: {
+    agentSafe?: Address;
+    agentEoas?: Address[];
+  };
+};
 
 /**
  * Hook for interacting with a single service.
  */
-export const useService = (service: Service) => {
-  // ChainIds used by the service
-  const chainIdsUsed = useMemo<number[]>(() => {
-    return Object.keys(service.chain_configs).map(Number);
-  }, [service.chain_configs]);
-
-  const addresses = useMemo<{
-    [chainId: number]: {
-      master: {
-        safe: Address;
-        signer: Address;
-      };
-      agent: {
-        safe: Address;
-        signer: Address;
+export const useService = ({
+  serviceConfigId = '',
+}: {
+  serviceConfigId?: string;
+}) => {
+  const { services, isLoaded } = useServices();
+  const queryClient = useQueryClient();
+
+  const service = useMemo(() => {
+    return services?.find(
+      (service) => service.service_config_id === serviceConfigId,
+    );
+  }, [serviceConfigId, services]);
+
+  const addresses: ServiceChainIdAddressRecord = useMemo(() => {
+    if (!service) return {};
+    const chainData = service.chain_configs;
+
+    // group multisigs by chainId
+    const addressesByChainId: ServiceChainIdAddressRecord = Object.keys(
+      chainData,
+    ).reduce((acc, chainIdKey) => {
+      const chainId = +chainIdKey;
+
+      const chain = chainData[chainId];
+      if (!chain) return acc;
+
+      const { multisig, instances } = chain.chain_data;
+
+      return {
+        ...acc,
+        [chainId]: {
+          agentSafe: multisig,
+          agentEoas: instances,
+        },
       };
-    };
-  }>(
-    () =>
-      chainIdsUsed.reduce((acc, chainId) => {
-        const chainConfig = service.chain_configs[chainId];
-        const master = {
-          safe: chainConfig.master.safe,
-          signer: chainConfig.master.signer,
-        };
-        const agent = {
-          safe: chainConfig.agent.safe,
-          signer: chainConfig.agent.signer,
-        };
-
-        return {
-          ...acc,
-          [chainId]: {
-            master,
-            agent,
-          },
-        };
-      }, {}),
-    [chainIdsUsed, service.chain_configs],
-  );
+    }, {});
+
+    return addressesByChainId;
+  }, [service]);
+
+  /**
+   * Overrides the deployment status of the service in the cache.
+   * @note Overwrite is only temporary if ServicesContext is polling
+   */
+  const setDeploymentStatus = (deploymentStatus?: MiddlewareDeploymentStatus) =>
+    queryClient.setQueryData(
+      REACT_QUERY_KEYS.SERVICE_DEPLOYMENT_STATUS_KEY(serviceConfigId),
+      deploymentStatus,
+    );
+
+  const deploymentStatus = queryClient.getQueryData<
```

Comment:
yeah, agreed..
this will be flaky without intervals 👎 

any suggestions on mutating/editing `data` from `useQuery`?

the reason i took this approach was to stay kinda consistent with `setDeploymentStatus` above, as an "override" to global consumers that rely on deploymentStatus until polling is reactivated

middleware endpoints return inconsistent deployment statuses (i.e. running, deploying, stopping) when we create a new service, causing the buttons to flicker between states -- creates bad ux

Reply:
will follow up in later PR as don't want to block services changes for now
have ur suggestions in dm!